var/spack/repos/builtin/packages/kokkos/package.py

```diff
@@ -62,12 +82,43 @@ def install(self, spec, prefix):
                 '--with-hwloc=%s' % spec['hwloc'].prefix,
                 '--with-serial'
             ]
+            # backends
             if '+openmp' in spec:
                 g_args.append('--with-openmp')
             if 'qthreads' in spec:
                 g_args.append('--with-qthreads=%s' % spec['qthreads'].prefix)
             if 'cuda' in spec:
                 g_args.append('--with-cuda=%s' % spec['cuda'].prefix)
+            # host architectures
+            host_arch = spec.variants['host_arch'].value
+
+            # gpu architectures
+            gpu_arch  = spec.variants['gpu_arch'].value
+
+            # only a host architecture
+            if (host_arch and not gpu_arch):
+                arch_args = '--arch=' + host_arch
+            # only a gpu architecture
+            if (gpu_arch and not host_arch):
+                if '+cuda' in spec:
+                    arch_args = '--arch=' + gpu_arch
+                else:
+                    raise InstallError(
+                        "Specified a GPU architecture %s without "
+                        "also specifying CUDA backend" %
+                        (gpu_arch))
```

Comment:
Can these be refactored into `conflicts()`?

Reply:
I think so. I will submit an update.